src/generate/peripheral.rs

```diff
@@ -104,75 +105,268 @@ pub fn render(
     Ok(out)
 }
 
+#[derive(Clone, Debug)]
 struct RegisterBlockField {
     field: syn::Field,
     description: String,
     offset: u32,
     size: u32,
 }
 
+#[derive(Clone, Debug)]
+struct Region {
+    fields: Vec<RegisterBlockField>,
+    offset: u32,
+    end: u32,
+}
+
+impl Region {
+    fn shortest_ident(&self) -> Option<String> {
```

Comment:
cc @jamesmunns

Reply:
I started out by integrating your code from the playground link and added some unit tests for it (there were a couple of minor issues, for example, the `dsr` case mentioned above would yield an empty identifier), but the conflicts were the limiting factor.   It's a shame that https://github.com/rust-lang/rfcs/blob/master/text/2102-unnamed-fields.md isn't available; it only just got merged and tracked in https://github.com/rust-lang/rust/issues/49804

Reply:
FWIW, I think it would be a shame if we forced the user to patch the SVD file to generate something usable.  I think its fine to be opinionated about naming and maintain a patched SVD, but requiring a patched SVD feels a bit heavyweight to me.

Reply:
One solution would be to have two implementations for naming, before code gen we use the `common_prefix` function and check that there is no name "collision" with other types and that a common prefix exists, if there aren't any problems carry on with code gen, otherwise use the current implementation (`shortest_ident`) for those types that are affected. I think this would satisfy us both.